raft/status.go

```diff
@@ -47,3 +49,20 @@ func getStatus(r *raft) Status {
 
 	return s
 }
+
+func (s Status) MarshalJSON() ([]byte, error) {
+	j := fmt.Sprintf(`{"id":"%x","term":%d,"vote":"%x","commit":%d, "lead":"%x","raftState":"%s","progress":{`,
```

Comment:
I see id, vote, and lead are hex strings. My concern is just that you are losing information when you convert the integer to the string (if the type is an integer, only numbers are valid. if the type is a string, anything is valid).